web/src/filter.ts

```diff
@@ -827,7 +828,7 @@ export class Filter {
             const conversation_terms = new Set(["channel", "topic", "dm"]);
             const filtered_terms = raw_terms.filter((term) => {
                 const operator = Filter.canonicalize_operator(term.operator);
-                return !conversation_terms.has(operator);
+                return !conversation_terms.has(operator) && operator !== "with";
```

Comment:
>For a link like:
http://localhost:9991/#narrow/stream/11-devel/topic/desktop/with/79
message_id is a DM ID. We narrow to the DM conversation but the with part is lost in the url. Is this intended ?

@prakhar1144 ya, it was intended, I added it here -- I thought like adding `with` to a `dm` narrow at this point wouldn't make sense since dm links are already permalinks. However, if you add `with` to a `dm` narrow now, they wouldn't be removed unlike previous, and would take you to the right narrow.